data/subscription.go

```diff
@@ -6,98 +6,48 @@ import (
-
-	// IsBundle tells us whether the subscription is a bundle
-	IsBundle bool `bson:"isBundle" json:"isBundle"`
-
-	// FromBundle defines if the app is inside a bundle and this event was triggered from a bundle
-	FromBundle bool `bson:"fromBundle" json:"fromBundle"`
-
-	// FromBundleID indicates which bundle id triggered this subscription event for the app
-	FromBundleID string `bson:"fromBundleId,omitempty" json:"fromBundleId,omitempty"`
-
-	// MigratedToBundle indicates if the subscription was migrated from a single app subscription to a bundle subscription
-	MigratedToBundle bool `bson:"migratedToBundle,omitempty" json:"migratedToBundle,omitempty"`
+	Type             events.EventObjectType `bson:"objectType" json:"objectType"`
+	App              events.App             `bson:"app" json:"app"`
+	Workspace        events.Workspace       `bson:"workspace" json:"workspace"`
+	Pricing          Pricing                `bson:"pricing" json:"pricing"`
+	Status           SubscriptionStatus     `bson:"status" json:"status"`
+	PeriodStart      time.Time              `bson:"periodStart" json:"periodStart"`
+	PeriodEnd        *time.Time             `bson:"periodEnd,omitempty" json:"periodEnd,omitempty"`
+	TerminationDate  *time.Time             `bson:"terminationDate,omitempty" json:"terminationDate,omitempty"`
+	IsSelfManaged    bool                   `bson:"isSelfManaged" json:"isSelfManaged"`
+	IsBundle         bool                   `bson:"isBundle" json:"isBundle"`
+	FromBundle       bool                   `bson:"fromBundle" json:"fromBundle"`
+	FromBundleID     string                 `bson:"fromBundleId,omitempty" json:"fromBundleId,omitempty"`
+	MigratedToBundle bool                   `bson:"migratedToBundle,omitempty" json:"migratedToBundle,omitempty"`
 }
 
-//#region PricingRecurrence
+// Pricing represents the pricing details for a subscription plan.
+type Pricing struct {
+	Name          string            `bson:"name,omitempty" json:"name,omitempty"`
+	Price         float32           `bson:"price" json:"price"`
+	IsUsageBased  bool              `bson:"isUsageBased,omitempty" json:"isUsageBased,omitempty"`
+	Recurrence    PricingRecurrence `bson:"recurrence" json:"recurrence"`
+	TrialEnabled  bool              `bson:"trialEnabled" json:"trialEnabled"`
+	TrialDuration int64             `bson:"trialDuration,omitempty" json:"trialDuration,omitempty"`
+}
 
-// PricingRecurrence defines different prices for monthly and yearly billing options
+// PricingRecurrence represents the recurrence options for pricing.
 type PricingRecurrence string
 
 const (
-	// MonthlyRecurrence const represents the string for monthly billing recurrence
-	MonthlyRecurrence PricingRecurrence = "monthly"
-
-	// YearlyRecurrence const represents the string for yearly billing recurrence
-	YearlyRecurrence PricingRecurrence = "yearly"
+	PricingRecurrenceMonthly PricingRecurrence = "monthly"
+	PricingRecurrenceYearly  PricingRecurrence = "yearly"
```

Comment:
Not sure if this part was on purpose?  But if anyone is consuming the package this rename would be a breaking change

Reply:
I think this part is good now